frontend/src/hooks/useObjectiveStatusMonitor.js

```diff
@@ -0,0 +1,45 @@
+import { useState, useCallback, useRef } from 'react';
+
+export default function useObjectiveStatusMonitor(objectives) {
+  const objectiveMap = useRef(
+    new Map(
+      // built to be invulnerable
+      (objectives || []) // missing param? no problem
+        .filter(({ ids }) => ids && Array.isArray(ids)) // missing prop? prop not an id? no problem
+        .map((o) => [JSON.stringify(o.ids.sort()), o]), // ok
+    ),
+  );
+
+  const [
+    atLeastOneObjectiveIsNotCompletedOrSuspended,
+    setAtLeastOneObjectiveIsNotCompletedOrSuspended,
+  ] = useState(
+    Array.from(objectiveMap.current.values()).some((o) => o.status !== 'Complete' && o.status !== 'Suspended'),
+  );
+
+  const dispatchStatusChange = useCallback((objectiveIds, localStatus) => {
+    try {
+      // same as before, we do not want white screens if something weird happens here
+      const objective = objectiveMap.current.get(JSON.stringify((objectiveIds || []).sort()));
+      if (!objective) {
+        return;
+      }
+
+      // update with the new status
+      objective.status = localStatus;
+
+      setAtLeastOneObjectiveIsNotCompletedOrSuspended(
+        Array.from(objectiveMap.current.values()).some((o) => o.status !== 'Complete' && o.status !== 'Suspended'),
```

Comment:
Should this go in a function as its used in two locations.

Reply:
sure, good idea